lib/Sema/BuilderTransform.cpp

```diff
@@ -694,8 +694,8 @@ class BuilderClosureRewriter
 
       // Load the right-hand side if needed.
       if (finalCapturedExpr->getType()->is<LValueType>()) {
```

Comment:
I think we should be using `hasLValueType()` here rather than `is<LValueType>()`, to account for tuples and paren types and such.

Reply:
Done!